File: frontend/src/types/posts.ts
/** 帖子 */
export interface PostItem {
  id: number;
  user_id: number;
  title: string;
  content: string;
  like_count: number;
  comment_count: number;
  status: number;
  created_at: string;
  updated_at: string;
  user_nickname: string | null;
  user_avatar: string | null;
  liked: boolean;
}

/** 帖子列表响应 */
export interface PostListResponse {
  items: PostItem[];
  offset: number;
  limit: number;
}

/** 创建帖子参数 */
export interface PostCreatePayload {
  title: string;
  content: string;
}

/** 更新帖子参数 */
export interface PostUpdatePayload {
  title?: string;
  content?: string;
}

/** 评论 */
export interface CommentItem {
  id: number;
  post_id: number;
  user_id: number;
  content: string;
  created_at: string;
  user_nickname: string | null;
  user_avatar: string | null;
}

/** 评论列表响应 */
export interface CommentListResponse {
  items: CommentItem[];
}

/** 创建评论参数 */
export interface CommentCreatePayload {
  content: string;
}

/** 点赞切换响应 */
export interface LikeToggleResponse {
  liked: boolean;
  like_count: number;
}

/** 点赞状态响应 */
export interface LikeStatusResponse {
  liked: boolean;
}

File: src/schema/posts.py
from typing import List, Optional

from pydantic import BaseModel


class PostResponse(BaseModel):
    id: int
    user_id: int
    title: str
    content: str
    like_count: int
    comment_count: int
    status: int
    created_at: str
    updated_at: str
    user_nickname: Optional[str] = None
    user_avatar: Optional[str] = None
    liked: bool = False


class PostListResponse(BaseModel):
    items: List[PostResponse]
    offset: int
    limit: int


class PostCreateRequest(BaseModel):
    title: str
    content: str


class PostUpdateRequest(BaseModel):
    title: str | None = None
    content: str | None = None
